examples/with-react-intl/components/PageWithIntl.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react'
+import {IntlProvider, addLocaleData, injectIntl} from 'react-intl'
+
+// Register React Intl's locale data for the user's locale in the browser. This
+// locale data was added to the page by `pages/_document.js`. This only happens
+// once, on initial page load in the browser.
+if (typeof window !== 'undefined' && window.ReactIntlLocaleData) {
```

Comment:
This whole way to pass locale data around seems quite contrived 🤔  As I commented in https://github.com/yahoo/react-intl/issues/731#issuecomment-279244388

Is there really no way to return the localeData from `getInitialProps` and simply pass that to `addLocaleData`?